addons/portal/controllers/mail.py

```diff
@@ -217,7 +80,7 @@ def portal_message_fetch(self, res_model, res_id, domain=False, limit=10, offset
         # Check access
         Message = request.env['mail.message']
         if kw.get('token'):
-            access_as_sudo = _check_special_access(res_model, res_id, token=kw.get('token'))
+            access_as_sudo = request.env[res_model]._get_thread_with_access(res_id, token=kw.get('token'))
```

Comment:
Strangely, this route doesn't support hash and pid?
Not a big deal as it was already like that before this PR, and it will be removed in future PR anyway.